tasks/omnibus.py

```diff
@@ -621,3 +623,58 @@ def rpath_edit(ctx, install_path, target_rpath_dd_folder, platform="linux"):
         if install_path in binary_rpath:
             new_rpath = os.path.relpath(target_rpath_dd_folder, os.path.dirname(file))
             _patch_binary_rpath(ctx, new_rpath, install_path, binary_rpath, platform, file)
+
+
+@task
+def deduplicate_files(ctx, directory):
+    def hash_file(filepath, chunk_size=65536):
+        """Returns the SHA-256 hash of the file's contents."""
+        hasher = hashlib.sha256()
+        with open(filepath, 'rb') as f:
+            while chunk := f.read(chunk_size):
+                hasher.update(chunk)
+        return hasher.hexdigest()
+
+    def find_duplicates(root_dir):
+        """Finds and returns duplicates as a map of hash -> list of files with that hash."""
+        hash_to_files = defaultdict(list)
+        for dirpath, _, filenames in os.walk(root_dir):
+            for name in filenames:
+                full_path = os.path.join(dirpath, name)
+                if not os.path.islink(full_path):  # Skip symlinks
```

Comment:
minor (unlikely to happen): consider also skipping sockets, fifos, device files, etc.
```suggestion
                if os.path.isfile(full_path) and not os.path.islink(full_path):  # Skip symlinks
```